File: src/main/java/dashflight/sparkbootstrap/PolicyCheckDao.java
package dashflight.sparkbootstrap;

import java.util.UUID;
import org.jdbi.v3.sqlobject.customizer.Bind;
import org.jdbi.v3.sqlobject.locator.UseClasspathSqlLocator;
import org.jdbi.v3.sqlobject.statement.SqlQuery;

@UseClasspathSqlLocator
public interface PolicyCheckDao {

    @SqlQuery
    Boolean checkUserPermission(@Bind("userId") UUID userId, @Bind("policy") String policy);

}

File: src/main/java/dashflight/sparkbootstrap/PolicyCheckDao.kt
package dashflight.sparkbootstrap

import org.jdbi.v3.sqlobject.customizer.Bind
import org.jdbi.v3.sqlobject.locator.UseClasspathSqlLocator
import org.jdbi.v3.sqlobject.statement.SqlQuery
import java.util.*

@UseClasspathSqlLocator
interface PolicyCheckDao {

    @SqlQuery
    fun getUserPolicies(@Bind("userId") userId: UUID?): Array<String>

    @SqlQuery
    fun checkIfUserIsAdmin(@Bind("userId") userId: UUID?): Boolean
}